### PR TITLE
Clean up the metric for pipeline duration

### PR DESCRIPTION
**Pull request by slim-bean**

### CODE REVIEW

File: pkg/logentry/pipeline.go [hunk 1]
```diff
@@ -71,7 +81,7 @@ func NewPipeline(logger log.Logger, stgs PipelineStages, plObserverMicroSeconds
 	return &Pipeline{
 		logger: log.With(logger, "component", "pipeline"),
 		stages: st,
-		plObs:  plObserverMicroSeconds,
+		plObs:  pipelineDuration.WithLabelValues(jobName),
```
Comment (Kuqd) [inline pkg/logentry/pipeline.go]: you don't need the observer.

File: pkg/logentry/pipeline.go [hunk 2]
```diff
@@ -82,11 +92,9 @@ func (p *Pipeline) Process(labels model.LabelSet, ts *time.Time, entry *string)
...
 		stage.Process(labels, ts, entry)
 	}
-	durUs := float64(time.Since(start).Nanoseconds()) / 1000
-	level.Debug(p.logger).Log("msg", "finished processing log line", "labels", labels, "time", ts, "entry", entry, "duration_us", durUs)
-	if p.plObs != nil {
-		(*p.plObs).Observe(durUs)
-	}
+	dur := time.Since(start).Seconds()
+	level.Debug(p.logger).Log("msg", "finished processing log line", "labels", labels, "time", ts, "entry", entry, "duration_s", dur)
+	p.plObs.Observe(dur)
```
Comment (Kuqd) [inline pkg/logentry/pipeline.go]:
```suggestion
	pipelineDuration.WithLabelValues(p.jobName).Observe(dur)
```

Comment (slim-bean) [reply pkg/logentry/pipeline.go]: Fixed!

File: pkg/logentry/pipeline.go [hunk 3]
```diff
@@ -7,24 +7,34 @@ import (
...
 	"github.com/grafana/loki/pkg/logentry/stages"
 	"github.com/grafana/loki/pkg/promtail/api"
 )
 
+var (
+	pipelineDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "logentry",
+		Name:      "pipeline_duration_seconds",
+		Help:      "Label and metric extraction pipeline processing time, in seconds",
+		Buckets:   []float64{.000005, .000010, .000025, .000050, .000100, .000250, .000500, .001000, .002500, .005000, .010000, .025000},
```
Comment (Kuqd) [inline pkg/logentry/pipeline.go]: prometheus.LinearBuckets or prometheus.ExponentialBuckets

Comment (slim-bean) [reply pkg/logentry/pipeline.go]: would be a little more concise, but for my brain I find it easier to just see the values, especially when dealing with microseconds

**Pull request merged**